tvm/src/codegen/hlsc/build_hlsc.cc

```diff
@@ -57,6 +60,7 @@ std::string BuildHLSC(
     case OutputMode::HostDevice : {code = cg.Finish(); break;}
     case OutputMode::HostOnly   : {code = cg.GetHost(); break;}
     case OutputMode::DeviceOnly : {code = cg.GetDevice(); break;}
+    case OutputMode::VHLSDevice : {code = cg.GetDevice(); break;}
```

Comment:
Looks redundant to me. The lat two modes actually do the things.

Reply:
It's just used to distinguish different backends. The current builder for debug mode only accepts two arguments, where the second one is the `OutputMode`.
https://github.com/cornell-zhang/heterocl/blob/e34f337bdd3350cab1d63e2dd0c38a2125c7867d/python/heterocl/tvm/build_module.py#L465-L475

We cannot generate other kernel codes other than OpenCL if still using these three output modes. `DeviceOnly` is misleading, since it only targets "sdaccel".

Reply:
HeteroCL returns only the device code in `DeviceOnly` mode. It should have nothing to do the with tool we are targeting. If you are using VHLS code generator, it will return VHLS no matter what mode it is using. That's weird.

Reply:
If you set the string `xcel`as `vhls` in 472 line: heterocl/python/heterocl/tvm/build_module.py, then HeteroCL should return the VHLS code.